sdk/core/azure-core/inc/azure/core/internal/hkeyholder.hpp

```diff
@@ -18,6 +20,9 @@
 
 #include <windows.h>
 
+#if !defined(WINAPI_PARTITION_DESKTOP) \
+    || WINAPI_PARTITION_DESKTOP // See azure/core/platform.hpp for explanation.
```

Comment:
What will happen on UWP if the whole type is under this if/def?

Reply:
t won't be available, unknown identifier. The same as with Http Transport Adapters.

Reply:
Correct me if I am mistaken, but does that mean we just fixed the UWP build issue, but would anyone running our SDK on UWP would end up with some runtime error or wrong behavior? If folks can't run it on that platform, do we need to support it?

Reply:
It won't compile, no runtime errors. Which is fine - UWP doesn't have registry, doesn't even have environment variables, so what would you use that class from, it would be useless on that platform, and that compilation issue is a good sign that you went into a code that you need to `#ifdef` for non-UWP case. Which is fine, it is exactly the same if you attempt to use this class on Linux, which also doesn't have registry.

Reply:
This is our telemetry output for UWP atm:
https://github.com/Azure/azure-sdk-for-cpp/blob/ae575b0c0838f73ef666270260c3c0385d097e78/sdk/core/azure-core/src/http/telemetry_policy.cpp#L67